src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -250,11 +250,13 @@ internal void ApplyHttpsDefaults(HttpsConnectionAdapterOptions httpsOptions)
 
     internal void ApplyDefaultCertificate(HttpsConnectionAdapterOptions httpsOptions)
     {
-        if (httpsOptions.ServerCertificate != null || httpsOptions.ServerCertificateSelector != null)
+        if (httpsOptions.HasServerCertificateOrSelector)
         {
             return;
         }
 
+        Debug.Assert(ApplicationServices.GetRequiredService<IHttpsConfigurationHelper>().IsEnabled);
```

Comment:
```suggestion
        Debug.Assert(ApplicationServices.GetRequiredService<IHttpsConfigurationHelper>().IsEnabled, "HTTPS must be enabled.");
```

Reply:
Personally, I don't usually add an assert message the location identifies the problem, but maybe this helps with telemetry or something?

Reply:
Debug.Assert will only show up during local development and CI runs. If this is an error case we think users can get to then it shouldn't be a Debug.Assert. If this is only an error case where we programmed something wrong in kestrel while working on some feature/bug fix, then Debug.Assert is good.

Reply:
It would indicate a programming error.  All current code paths satisfy the assert.

Reply:
I assume you brought it up because of the comment.  It was intended as guidance to whoever has to debug when we introduce that programming error in the future.  Happy to reword if it's unclear.

Reply:
Rephrased.